### PR TITLE
pkg/api/testing: add deepcopy smoke test to roundtrip test

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/kubernetes/issues/41438.

### CODE REVIEW

**Comment by sttts** [comment]
This probably need some extra cases for certain types which are not deep-copied intentionally (time.Time?).

**Comment by sttts** [comment]
And it does not detect when the deep-copy machinery switches to the reflection based fallback code path.

File: staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip.go [hunk 1]
```diff
@@ -232,6 +232,15 @@ func roundTrip(t *testing.T, scheme *runtime.Scheme, codec runtime.Codec, object
 		t.Errorf("3: %v: diff: %v\nCodec: %#v", name, diff.ObjectReflectDiff(object, obj3), codec)
 		return
 	}
+
+	// do structure-preserving fuzzing of the deep-copied object. If it shares anything with the original,
+	// the deep-copy was actually only a shallow copy. Then original and obj3 will be different after fuzzing.
+	// NOTE: we use the encoding+decoding here as an alternative, guaranteed deep-copy to compare against.
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/api/testing/roundtrip.go]: Clever.

**Comment by deads2k** [comment]
lgtm.  please squash bazel.

**Pull request merged**